plugins/woocommerce/includes/admin/class-wc-admin-reports.php

```diff
@@ -211,22 +213,55 @@ public static function get_reports() {
 		 *         if ( is_admin() ) {
 		 *             // ...
 		 *
+		 * @since 2.1.0
+		 *
 		 * @param array $reports The associative array of reports.
 		 */
-		$reports = apply_filters( 'woocommerce_admin_reports', $reports );
-		// Backwards compatibility.
-		$reports = apply_filters_deprecated(
+		$filtered_reports = apply_filters( 'woocommerce_admin_reports', $reports );
+
+		/*
+		 * Check if there is any use of the legacy `woocommerce_admin_reports` filter to send a deprecation warning.
+		 * We will remove non-compliant entries twice.
+		 * First, here to check if there are any reports-specific, non-analytics changes to the original array.
+		 * Then send the non-sanitized array to the `woocommerce_reports_charts` filter to ensure 100% backward compatibility.
+		 * Then, we will sanitize again the result of both filters.
+		 */
+		$filtered_legacy_reports = $filtered_reports;
```

Comment:
As the code is now if we'd loop through `$filtered_reports` first and then make a copy we wouldn't need to unset a second time when looping through the reports. But if you refactor as suggested in another comment then it might still need to be done twice.